### ISSUE TITLE
Unable to produce clustering

### ISSUE THREAD

**Issue by lykhahaha** [VincentCheungM/Run_based_segmentation#1]
Hi VincentCheungM , thanks for all your works. However, I have encountered some problems when running your code.
(1) Unable to produce clustering
My running environment is ubuntu14.04 and ros indigo.

my data : 2011_09_30_drive_18.bag from kitti.( 64 velodyne)

my cofig :
sensor_model_, 64
th_run_, 0.5
th_merge_, 0.1
OR
sensor_model_, 64
th_run_, 0.8
th_merge_, 0.2
There is no clustering here, no matter how I modify the parameters。

(2) Failed to find match for field 'ring'.
when I rosrun points_preprocessor_usi groundplanfit,The prompt will be printed on the screen：
“Failed to find match for field 'ring'.”　However, This does not affect ground segmentation.
This node will still be able to divide the ground well.

Thank you again for your work and looking forward to your reply.

**Comment by VincentCheungM** [comment]
Yes, it's mainly due to the process in [here](https://github.com/VincentCheungM/Run_based_segmentation/blob/master/nodes/clustering/scanlinerun.cpp#L393-#L404). 

```cpp
    // Organize Pointcloud in scanline
    SLRPointXYZIRL point;
    for(int i=0;i<laserCloudIn.points.size();i++){
        point.x = laserCloudIn.points[i].x;
        point.y = laserCloudIn.points[i].y;
        point.z = laserCloudIn.points[i].z;
        point.intensity = laserCloudIn.points[i].intensity;
        point.ring = laserCloudIn.points[i].ring;
        point.label = 0u;// 0 means uncluster

        if(point.ring<=sensor_model_&&point.ring>=0){
            laserCloudScans_[point.ring]->push_back(point);
        }    
    }
```

It's re-organizing the points within point cloud to the correspond scan line. The `ring` is the scan line index of the points, which can be achieved with the help of Velodyne drive. However, the binary point cloud files from KITTI don't provide the `ring` property, and it goes wrong. So you can calculate the ring by your self or using other point cloud files.
The `ring` can be calculated in this formula(latex code):

$$ring =\arcsin{\frac{z}{\sqrt{x^2+y^2+z^2}}}$ $

**Comment by lykhahaha** [comment]
Thank you for your help.I also think the reason for the failure of clustering is because of the lack of rings.
The result of this formula is angle(float), but the value of rings should be 0 to 63 (int). How should I convert?

**Comment by VincentCheungM** [comment]
Maybe `round` is enough.
